### ISSUE TITLE
freeze_graph not working in v0.11.0 . giving V2 error

### ISSUE THREAD

**Issue by dhwajraj** [tensorflow/tensorflow#6032]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?
https://github.com/tensorflow/tensorflow/issues/5938
https://stackoverflow.com/questions/40616430/load-exported-model-in-tensorflow
https://github.com/tensorflow/tensorflow/issues/5639
https://stackoverflow.com/questions/40616430/load-exported-model-in-tensorflow

### Environment info
Operating System: CentOS (AWS p2.xlarge)

Installed version of CUDA and cuDNN:  cuda 8.0 , cudnn 5.1.5
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):
```
/usr/local/cuda-8.0/lib64/libcudadevrt.a
/usr/local/cuda-8.0/lib64/libcudart.so
/usr/local/cuda-8.0/lib64/libcudart.so.8.0
/usr/local/cuda-8.0/lib64/libcudart.so.8.0.44
/usr/local/cuda-8.0/lib64/libcudart_static.a
/usr/local/cuda-8.0/lib64/libcudnn.so
/usr/local/cuda-8.0/lib64/libcudnn.so.5
/usr/local/cuda-8.0/lib64/libcudnn.so.5.1.5
/usr/local/cuda-8.0/lib64/libcudnn_static.a
```

If installed from binary pip package, provide:

1. A link to the pip package you installed:
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`. 
```
0.11.0
```
If installed from source, provide 

1. The commit hash (`git rev-parse HEAD`)
```
https://github.com/tensorflow/tensorflow/archive/v0.11.0.zip
```
2. The output of `bazel version`
```
Build label: 0.3.2
Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Fri Oct 7 17:25:10 2016 (1475861110)
Build timestamp: 1475861110
Build timestamp as int: 1475861110
```

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
I trained my convolution network on v0.11.0 and then running freeze_graph on the same tensorflow v0.11.0 . freeze_graph does not work.
```
saving model:
saver = tf.train.Saver(tf.all_variables(), max_to_keep=100)
saver.save(sess, checkpoint_prefix, global_step=current_step)

writing graph:
tf.train.write_graph(sess.graph.as_graph_def(), checkpoint_prefix, "graph"+str(nn)+".pb", as_text=False)

running freeze_graph:
tensorflow-0.11.0/bazel-bin/tensorflow/python/tools/freeze_graph --input_graph=graph4000 --input_checkpoint=runs/1479444509/checkpoints/model-4000 --output_graph=my.pb --output_node_names=output0/predictions0,accuracy0/accuracy0,output0/scores0,output0/Softmax --input_binary=True --clear_devices=True --initializer_nodes=embedding/W

```

### Logs or other output that would be helpful
```
tensorflow-0.11.0/bazel-bin/tensorflow/python/tools/freeze_graph --input_graph=graph4000 --input_checkpoint=runs/1479444509/checkpoints/model-4000 --output_graph=my.pb --output_node_names=output0/predictions0,accuracy0/accuracy0,output0/scores0,output0/Softmax --input_binary=True --clear_devices=True --initializer_nodes=embedding/W
Traceback (most recent call last):
  File "/data3/tensorflow-0.11.0/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/tools/freeze_graph.py", line 134, in <module>
    tf.app.run()
  File "/data3/tensorflow-0.11.0/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/platform/app.py", line 30, in run
    sys.exit(main(sys.argv[:1] + flags_passthrough))
  File "/data3/tensorflow-0.11.0/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/tools/freeze_graph.py", line 131, in main
    FLAGS.output_graph, FLAGS.clear_devices, FLAGS.initializer_nodes)
  File "/data3/tensorflow-0.11.0/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/tools/freeze_graph.py", line 103, in freeze_graph
    _ = tf.import_graph_def(input_graph_def, name="")
  File "/data3/tensorflow-0.11.0/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/framework/importer.py", line 258, in import_graph_def
    op_def = op_dict[node.op]
KeyError: u'RestoreV2'
```

**Comment by poxvoculi** [comment]
My best guess is some kind of version skew, similar to the suggestion for #5639.  @petewarden might be the best person for advice, but he's very over-burdened with requests at the moment.

Your immediate problem is a node name not being in the op_dict.  If I were debugging for myself, I'd try to figure out whether it's just this one node missing, or maybe you somehow failed to get all of the definitions for the graph you're trying to restore.

**Comment by petewarden** [comment]
I have a code change pending to make the error here a bit more informative. I believe the issue is what @poxvoculi describes though, that you trained the model in 0.12 but are trying to load it in 0.11. I'm closing for now on that assumption, but please re-open and let me know if I'm incorrect!

**Comment by dhwajraj** [comment]
I am pretty sure that I trained the model in 0.11 and freezing it using 0.11 only. Anyway I think whatever the issue was, is solved in 0.10 and 0.12rc0.
